### PR TITLE
IRMutator->IRMutator2, Part 4

### PR DESCRIPTION
**Pull request by steven-johnson**
Grab bag of conversions (split into multiple PRs for ease of review)

### CODE REVIEW

File: src/VectorizeLoops.cpp [hunk 1]
```diff
@@ -727,26 +717,30 @@ class VectorSubs : public IRMutator {
...
                                          predicated_stmt);
                     debug(4) << "...Predicated IfThenElse: \n" << stmt << "\n";
+                    return stmt;
                 }
             } else {
                 // It's some arbitrary vector condition.
                 if (!vectorize_predicate) {
-                    debug(4) << "...Scalarizing vector predicate: \n" << stmt << "\n";
-                    stmt = scalarize(op);
+                    Stmt stmt = scalarize(op);
```
Comment (abadams) [inline src/VectorizeLoops.cpp]: This changes the order in which debugging messages are printed.

Comment (steven-johnson) [reply src/VectorizeLoops.cpp]: True, but 'stmt' was undefined at this point in the old code, so this printed nothing.

Comment (abadams) [reply src/VectorizeLoops.cpp]: I think that issue is independent -- I was thinking that if there are debugging prints done inside scalarize(...), then in the old code these gets printed after "scalarizing vector predicate...", and in the new code it gets printed before that. But the stmt arg should indeed have been something like Stmt(op). I think printing op directly might print it as a pointer.

Comment (steven-johnson) [reply src/VectorizeLoops.cpp]: You are correct. Fixed.

**Comment by steven-johnson** [comment]
Did this have an implicit LGTM-if-the-one-issue-was-fixed?

**Pull request merged**